src/inmanta/data/__init__.py

```diff
@@ -3938,6 +3938,24 @@ def _get_paging_history_item_count_query(
 
         return sql_query, values
 
+    @classmethod
+    async def get_resource_deploy_summary(cls, environment: uuid.UUID) -> Dict[str, int]:
+        query = f"""
+            SELECT COUNT(res.resource_id) as count, status
+            FROM
+            (SELECT DISTINCT ON (r.resource_id) r.resource_id, r.status
+                FROM {cls.table_name()} as r
```

Comment:
How does this restrict the results to the last released version? 
i.e. doesn't this mix resources from older versions in?

Reply:
It takes the last released version of each resource. If there is a higher version, and the resource is not included in it, then that would be an `orphaned` resource, but I seem to remember that we discussed that we wouldn't show them...

Ah, ok I think I've got it, the orphaned ones shouldn't even be in the summary, right?
I'll implement the changes

Reply:
The query restricts everything to the latest version by a combination of `SELECT DISTINCT` and the ordering `ORDER BY r.resource_id, r.model DESC)`. Mixing up versions is not possible because this query runs in a transaction and updating `configurationmodel.released` is also done transactional. But I have to admit that it's hard to read. Maybe we need to add some more comments to the query.